cmd/tink-worker/container/container.go

```diff
@@ -0,0 +1,169 @@
+}
+
+type removeOpts struct {
+	force   bool
+	links   bool
+	volumes bool
+}
+
+// Create creates container and returns containerid on success else returns error.
+func (c cdata) Create(ctx context.Context, cli client.ContainerAPIClient) (string, error) {
+	fmt.Println(c.data)
+	if c.data.name == "" || c.data.imageName == "" {
+		return "", errors.New("container name and image name are required data")
+	}
+
+	if cli == nil || (reflect.ValueOf(cli).IsNil()) {
+		return "", errors.New("ContainerAPIClient interface is not valid")
+	}
+
+	if len(c.data.cmd) == 0 {
+		return "", errors.New("cmd has invalid data")
+	}
+
+	if len(c.data.volumeBindings) > 0 {
+		isValidVolume := regexp.MustCompile(`^[a-zA-Z0-9/:-]*$`).MatchString
+		for _, value := range c.data.volumeBindings {
+			if !isValidVolume(value) {
+				return "", errors.New("volume has invalid data")
+			}
+		}
+	}
+
+	config := &container.Config{
+		Image:        path.Join(c.data.registry, c.data.imageName),
+		AttachStdout: true,
+		AttachStderr: true,
+		Cmd:          c.data.cmd,
+		Tty:          true,
+		Env:          c.data.env,
+	}
+	if !c.data.captureLogs {
+		config.AttachStdout = false
+		config.AttachStderr = false
+		config.Tty = false
+	}
+
+	hostConfig := &container.HostConfig{
+		Privileged: true,
+		Binds:      c.data.volumeBindings,
+	}
+
+	if pidConfig := c.data.pid; pidConfig != "" {
+		hostConfig.PidMode = container.PidMode(pidConfig)
+	}
+
+	resp, err := cli.ContainerCreate(ctx, config, hostConfig, nil, nil, c.data.name)
+	if err != nil {
+		return "", errors.Wrap(err, "DOCKER CREATE")
+	}
+	return resp.ID, nil
+}
+
+// Start starts container and returns error on failure.
+func (c cdata) Start(ctx context.Context, cli client.ContainerAPIClient, id string) error {
+	if id == "" {
+		return errors.New("container id is not valid")
+	}
+
+	if cli == nil || (reflect.ValueOf(cli).IsNil()) {
+		return errors.New("ContainerAPIClient interface is not valid")
+	}
+
+	err := cli.ContainerStart(ctx, id, types.ContainerStartOptions{})
+	if err != nil {
+		return errors.Wrap(err, "DOCKER START")
+	}
+	return nil
+}
+
+// Remove delete container and remove volumes, links if flags are set to true. returns error on failure.
+func (c cdata) Remove(ctx context.Context, cli client.ContainerAPIClient, id string, opts removeOpts) error {
+	if id == "" {
+		return errors.New("container id is not valid")
+	}
+
+	if cli == nil || (reflect.ValueOf(cli).IsNil()) {
+		return errors.New("ContainerAPIClient interface is not valid")
+	}
+
+	// create options for removing container
+	options := types.ContainerRemoveOptions{
+		Force:         opts.force,
+		RemoveLinks:   opts.links,
+		RemoveVolumes: opts.volumes,
+	}
+
+	// send API call to remove the container
+	err := cli.ContainerRemove(ctx, id, options)
+	if err != nil {
+		return errors.Wrap(err, "DOCKER REMOVE")
+	}
+	return nil
+}
+
+// Wait waits on any non "not-running" states and returns container state on success or returns error on failure.
```

Comment:
sorry, this double negative is difficult to understand. is a `non "not-running"` state just a running state?